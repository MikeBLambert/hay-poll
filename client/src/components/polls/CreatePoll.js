import React, { Component } from 'react'
import PropTypes from 'prop-types'
import PollPreview from './PollPreview';

export default class CreatePoll extends Component {
  static propTypes = {
  }

  state = {
    title: '',
    option: '',
    options: []
  };

  updateOption = ({ target }) => {
    this.setState({ option: target.value });
  };

  updateTitle = ({ target }) => {
    this.setState({ title: target.value });
  };


  updateOptionsArray = () => {
    const newOption = this.state.options.concat(this.state.option);
    this.setState({ options: newOption });

  }

  render() {
    const { title, option, options } = this.state;
    return (
      <div>
        <h1>Create Poll!</h1>

        <label>Add Title</label>
        <input type="text" value={title} onChange={this.updateTitle} />

        <label>Add Option</label>
        <input type="text" value={option} onChange={this.updateOption} />
        <button onClick={this.updateOptionsArray}>Add</button>
        <PollPreview title={title} options={options} />
      </div>
    );
  }
}
