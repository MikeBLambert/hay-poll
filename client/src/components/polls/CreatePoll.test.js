import React from 'react';
import CreatePoll from './CreatePoll';
import { shallow } from 'enzyme';

describe('CreatePoll component', () => {
  it('renders a component', () => {
    const wrapper = shallow(<CreatePoll />);
    expect(wrapper).toMatchSnapshot();
  })
})
