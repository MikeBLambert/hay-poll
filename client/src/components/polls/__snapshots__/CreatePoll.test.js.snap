// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoll component renders a component 1`] = `
<div>
  <h1>
    Create Poll!
  </h1>
  <label>
    Add Option
  </label>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
